### DISCUSSION TITLE
Code walk through

### DISCUSSION THREAD

**Discussion by knapsea** [nasa/fprime#848] [Q&A] [answered]
I been see documentation on concepts and have these queries
1) Once a project is built using fprime-util generate , and development has commenced on the initial defined components, how easy would it add a new component  later in development?
2) Is there a document I can follow to see the main loop, essential components needed for a desktop application ?

Thanks
G

**Comment by Joshua-Anderson** [comment, marked as answer]
> Once a project is built using fprime-util generate , and development has commenced on the initial defined components, how easy would it add a new component later in development?

Create a new component, add it to the CMakeLists.txt file of the parent directory, and you should be good to go!
[this](https://nasa.github.io/fprime/v2.0.0/UsersGuide/best/development-practice.html) is brief overview of what the development workflow looks like.

> Is there a document I can follow to see the main loop, essential components needed for a desktop application ?

I'd recommend taking a look at the Ref app as a baseline example for a desktop application. It does have a few extra components for demonstration purposes (all the components in the Ref:: namespace, primarily the signalgen component instances), but they can be easily removed.

**Reply by knapsea** [reply]
Thanks!
